bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ValidateProjectEncoding.java

```diff
@@ -95,6 +99,22 @@ static void updateMissingEncodingMarker(IProject project) {
 		}
 	}
 
+	/**
+	 * Returns the current severity for
+	 * {@link ResourcesPlugin#PREF_MISSING_ENCODING_MARKER_SEVERITY}.
+	 *
+	 * @return current severity. Default value is nothing else is specified.
+	 */
+	private static int getSeverity() {
+		int severity = PreferenceInitializer.PREF_MISSING_ENCODING_MARKER_SEVERITY_DEFAULT;
+		IEclipsePreferences node = InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES);
```

Comment:
I think this will not work with product customization (where one could set it once for product and never set INSTANCE value). So better for **reading** the value use 

`Platform.getPreferencesService().getInt(ResourcesPlugin.PI_RESOURCES,					PREF_MISSING_ENCODING_MARKER_SEVERITY, severity, null);`

**Writing** goes as before to the instance scope.

Reply:
replaced reading w/ pref service access

Reply:
Btw: I had taken that approach from class ``CheckMissingNaturesListener::getMissingNatureSeverity`` (as suggested when I started this issue). Maybe we should create a ticket for fixing that class, too.